cpp/tests/experimental/spatial/spatial_window_test.cu

```diff
@@ -0,0 +1,121 @@
+
+  EXPECT_EQ(expected_points, result_points);
+}
+
+TYPED_TEST(SpatialWindowTest, SimpleTest)
+{
+  using T      = TypeParam;
+  using Vec    = cuspatial::vec_2d<T>;
+  using VecVec = std::vector<Vec>;
+  auto points  = rmm::device_vector<Vec>(VecVec({{1.0, 0.0},
+                                                {2.0, 1.0},
+                                                {3.0, 2.0},
+                                                {5.0, 3.0},
+                                                {7.0, 1.0},
+                                                {1.0, 3.0},
+                                                {2.0, 5.0},
+                                                {3.0, 6.0},
+                                                {6.0, 5.0},
+                                                {0.0, 4.0},
+                                                {3.0, 7.0},
+                                                {6.0, 4.0}}));
+
+  auto expected_points = rmm::device_vector<Vec>(VecVec({{3.0, 2.0}, {5.0, 3.0}, {2.0, 5.0}}));
+
+  auto result_size = cuspatial::count_points_in_spatial_window(
+    Vec{1.5, 1.5}, Vec{5.5, 5.5}, points.begin(), points.end());
+
+  EXPECT_EQ(result_size, expected_points.size());
+
+  auto result_points = rmm::device_vector<Vec>(result_size);
+  cuspatial::points_in_spatial_window(
+    Vec{1.5, 1.5}, Vec{5.5, 5.5}, points.begin(), points.end(), result_points.begin());
+
+  EXPECT_EQ(expected_points, result_points);
+}
+
+// Test that windows with min/max reversed still work
+TYPED_TEST(SpatialWindowTest, ReversedWindow)
+{
+  using T      = TypeParam;
+  using Vec    = cuspatial::vec_2d<T>;
+  using VecVec = std::vector<Vec>;
+  auto points  = rmm::device_vector<Vec>(VecVec({{1.0, 0.0},
+                                                {2.0, 1.0},
+                                                {3.0, 2.0},
+                                                {5.0, 3.0},
+                                                {7.0, 1.0},
+                                                {1.0, 3.0},
+                                                {2.0, 5.0},
+                                                {3.0, 6.0},
+                                                {6.0, 5.0},
+                                                {0.0, 4.0},
+                                                {3.0, 7.0},
+                                                {6.0, 4.0}}));
+
+  auto expected_points = rmm::device_vector<Vec>(VecVec({{3.0, 2.0}, {5.0, 3.0}, {2.0, 5.0}}));
+
+  auto result_size = cuspatial::count_points_in_spatial_window(
+    Vec{5.5, 5.5}, Vec{1.5, 1.5}, points.begin(), points.end());
+
+  EXPECT_EQ(result_size, expected_points.size());
+
+  auto result_points = rmm::device_vector<Vec>(result_size);
+  cuspatial::points_in_spatial_window(
+    Vec{5.5, 5.5}, Vec{1.5, 1.5}, points.begin(), points.end(), result_points.begin());
+
+  EXPECT_EQ(expected_points, result_points);
+}
```

Comment:
Suggest two more tests: one for "all points in window" which is trivial. The other is less trivial - do we need to test edge cases, like points along the edge? What level of precision will it work/fail? That's a great thing to cover in test cases.

Reply:
Good ideas. Added new tests and cleaned up the tests (and factored out repetitive code).

Reply:
Did you create a new file for the new tests? I see one new test, `Empty`, and that you renamed the file from `.cpp` to `.cu`, but I don't see an all points or edge test.

Reply:
I see them, looking good!